src/main/clojure/clara/rules/compiler.clj

```diff
@@ -235,15 +235,21 @@
   "Compiles a sequence of constraints into a structure that can be evaluated.
 
    Callers may also pass a collection of equality-only-variables, which instructs
-   this function to only do an equality check on them rather than create a unification binding."
+   this function to only do an equality check on them rather than create a unification binding.
+
+   The returned expression evaluates to a two item vector, [bindings error], which contains
+   the evaluated bindings in the first argument and a map with information on any exception
+   that might have been thrown by evaluating a constraint as the second vector entry."
```

Comment:
It looks like Closure, which is used by ClojureScript and I believe is ultimately the source of these warnings, supports suppressing them in particular parts of JavaScript code.  Perhaps ClojureScript has some support for passing such annotations through?  I couldn't immediately find anything on warning suppression in ClojureScript but it looks like it can pass them through for type annotation: https://github.com/clojure/clojurescript/wiki/Compile-Time-Type-Checking

The problem I see with these warnings is that they'll come up in user code consuming the cljs version of Clara, correct?.  If it were just our build that was the issue it would be simple enough to just replace functions in constraints in the cljs tests that can't throw with ones that can.